x-pack/plugins/apm/public/components/shared/charts/TransactionCharts/index.tsx

```diff
@@ -150,78 +165,89 @@ export class TransactionCharts extends Component<TransactionChartProps> {
-    const { responseTimeSeries, tpmSeries } = charts;
-    const { transactionType } = urlParams;
-    const maxY = getMaxY(responseTimeSeries);
-    const formatter = getDurationFormatter(maxY);
+  const { responseTimeSeries, tpmSeries } = charts;
+  const { transactionType } = urlParams;
+  const [maxY, setMaxY] = useState(0);
+  useEffect(() => {
+    setMaxY(getMaxY(responseTimeSeries));
+  }, [responseTimeSeries]);
 
-    return (
-      <>
-        <EuiFlexGrid columns={2} gutterSize="s">
-          <EuiFlexItem data-cy={`transaction-duration-charts`}>
-            <EuiPanel>
-              <React.Fragment>
-                <EuiFlexGroup justifyContent="spaceBetween">
-                  <EuiFlexItem>
-                    <EuiTitle size="xs">
-                      <span>{responseTimeLabel(transactionType)}</span>
-                    </EuiTitle>
-                  </EuiFlexItem>
-                  <LicenseContext.Consumer>
-                    {(license) =>
-                      this.renderMLHeader(license?.getFeature('ml').isAvailable)
-                    }
-                  </LicenseContext.Consumer>
-                </EuiFlexGroup>
-                <TransactionLineChart
-                  series={responseTimeSeries}
-                  tickFormatY={getResponseTimeTickFormatter(formatter)}
-                  formatTooltipValue={getResponseTimeTooltipFormatter(
-                    formatter
-                  )}
-                />
-              </React.Fragment>
-            </EuiPanel>
-          </EuiFlexItem>
+  const averageY = getAverageY(responseTimeSeries);
 
-          <EuiFlexItem style={{ flexShrink: 1 }}>
-            <EuiPanel>
-              <React.Fragment>
-                <EuiTitle size="xs">
-                  <span>{tpmLabel(transactionType)}</span>
-                </EuiTitle>
-                <TransactionLineChart
-                  series={tpmSeries}
-                  tickFormatY={this.getTPMFormatter}
-                  formatTooltipValue={this.getTPMTooltipFormatter}
-                  truncateLegends
-                />
-              </React.Fragment>
-            </EuiPanel>
-          </EuiFlexItem>
-        </EuiFlexGrid>
-        {transactionType === TRANSACTION_PAGE_LOAD && (
-          <>
-            <EuiSpacer size="s" />
-            <EuiFlexGrid columns={2} gutterSize="s">
-              <EuiFlexItem>
-                <EuiPanel>
-                  <DurationByCountryMap />
-                </EuiPanel>
-              </EuiFlexItem>
-              <EuiFlexItem>
-                <EuiPanel>
-                  <BrowserLineChart />
-                </EuiPanel>
-              </EuiFlexItem>
-            </EuiFlexGrid>
-          </>
-        )}
-      </>
-    );
-  }
+  const formatTooltip = getDurationFormatter(averageY ?? maxY);
+  const formatter = getDurationFormatter(maxY);
+
+  const onToggleLegend = (series: Serie[]) => {
+    if (!isEmpty(series)) {
+      setMaxY(getMaxY(series as TimeSeries[]));
```

Comment:
I have some concerns here, mostly around us having to maintain state about CustomPlot in TransactionCharts, which could easily get out of sync. For instance, this happens when I enter a filter into the query bar that gives no results, and then go back:

![image](https://user-images.githubusercontent.com/352732/90239713-4a9bbd00-de28-11ea-9427-ec6e88ba9206.png)

The axis unit is not updated. Of course we can probably fix this even with this implementation, but it might be better to consider having state only one place (for instance, the consumers of `CustomPlot`.

Reply:
@dgieselaar I've rethought the solution to this problem. I removed the state (https://github.com/elastic/kibana/pull/74931/commits/a04aabac22cb0b9735600c0b2e05754b24b0eab9#diff-da514a7cc4e8210906a22c7f824eaea9L170-L184) and I use `asDuration` to calculate what will be the unit shown in the y-axis based on the value passed by the `CustomPlot`. This fixes the issue of this ticket and also the bug you reported.